torch/_inductor/autotune_process.py

```diff
@@ -171,22 +171,42 @@ def put(self, obj: Any) -> None:
         assert self.request_queue is not None
         self.request_queue.put(obj)
 
-    def get(self) -> Any:
+    def get(
+        self, result_timeout=1.0, graceful_timeout=3.0, terminate_timeout=1.0
```

Comment:
How is the timeout chosen?  What about very large problem sizes?

Reply:
They are configured manually and passed through further below from these (new) inductor configuration settings: max_autotune_subproc_result_timeout_seconds, max_autotune_subproc_graceful_timeout_seconds, max_autotune_subproc_terminate_timeout_seconds.

Of these, the max_autotune_subproc_result_timeout_seconds is the most important one as that one specifies how long we allow Kernels to run before we try to stop them. I gave it a default value of 30 seconds. The others are just additional timeouts that specify how long after a graceful shutdown we proceed by sending a SIGTERM signal, and if even that is not sufficient, a SIGKILL.

Since we are talking about actual Kernel runtime, it's my experience that 30 seconds are more than enough for any GEMM that I can fit in real-world GPU memory to finish unless it's somehow broken. 

But, that said, in order not to change default behavior and to be on the safe side, we could of course start with a much higher or even unlimited default threshold. What do you think?

Reply:
I made the default timeouts longer and added a proper doc comment in case any code outside of TuningProcessPool starts to use it ( I did not find any usage beyond that ).